server/bot.go

```diff
@@ -50,3 +52,33 @@ func (p *Plugin) PostBotCustomDM(userID string, message string, todo string, iss
 
 	return appError
 }
+
+// ReplyPostBot post a message and a todo in the same thread as the post postID
+func (p *Plugin) ReplyPostBot(postID, message, todo string) error {
+	if postID == "" {
+		return errors.New("Post ID not defined")
+	}
+
+	post, appErr := p.API.GetPost(postID)
+	if appErr != nil {
+		return appErr
+	}
+	rootID := post.Id
+	if post.RootId != "" {
+		rootID = post.RootId
+	}
+
+	quotedTodo := "\n> " + strings.Join(strings.Split(todo, "\n"), "\n> ")
+	_, appErr = p.API.CreatePost(&model.Post{
+		UserId:    p.BotUserID,
+		ChannelId: post.ChannelId,
+		Message:   message + quotedTodo,
+		RootId:    rootID,
```

Comment:
Sorry for my own continued confusion, but should we also be setting the ParentID on this post?

Reply:
After checking the code and asking in the [Server channel](https://community.mattermost.com/core/pl/uezrbne3utg6bxq9c365wbmndy), RootID and ParentID should have the same ID, and the CreatePost API makes sure that if no ParentID is given, use RootID. If you want I can be verbose here and set parentID to rootID too, but I think it is not needed.